hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java

```diff
@@ -628,8 +628,10 @@ private static void queueXmlGenerator(XmlGenerator gen, Config config) {
             QueueStoreConfig storeConfig = q.getQueueStoreConfig();
             if (storeConfig != null) {
                 gen.open("queue-store", "enabled", storeConfig.isEnabled())
-                        .node("class-name", storeConfig.getClassName())
-                        .node("factory-class-name", storeConfig.getFactoryClassName())
+                        .node("class-name",
+                            classNameOrImplClass(storeConfig.getClassName(), storeConfig.getStoreImplementation()))
```

Comment:
I have serious concerns about the method `classNameOrImplClass`. I understand that it was introduced before, but it leads to inconsistent configuration. Consider that I set an instance `new MyClass().setMyField("value")`. What happens with `classNameOrImplClass`, is that the class name is extracted, while the rest context is lost. It means after I deserialize the configuration back, I'll get one of the two issues:
1) Exception - in case there is no default constructor 
2) Lost value of the `myField`

So my question is why do we rely on this inconsistent behavior in the first place? From what I see, this XML configuration is only used for Management Center, is it correct? If yes, then how exactly MC use it? Are we ok with giving that incorrect configuration to the user?
@puzpuzpuz @mmedenjak I would appreciate your comments.

Reply:
I doubt that `queue-store` element is only used for Management Center. But as for `management-center` config element, it's going to change a lot in #16161.

Reply:
My question was about the general approach since there are multiple usages of the method in question. From what I see in the code, `ConfigXmlGenerator` is used only in `MemberConfigRequest` and `GetMemberConfigMessageTask`, which are both related to MC. 
So that if we know for sure that MC is the only real consumer of `ConfigXmlGenerator` result and MC users are informed about that potential inconsistency, then we may leave the current approach.
Otherwise, I am not sure if it is viable given the surprising results which users may get.

Reply:
> So that if we know for sure that MC is the only real consumer of ConfigXmlGenerator result

I believe that it's the case. As for the use case in MC, we don't deserialize all parts of the config, only some of them. And parts that we deserialize don't include user defined classes.

Adding @emre-aydin into the loop.

Reply:
To the best of my knowledge, MC is the only consumer of this info.

Reply:
@puzpuzpuz @emre-aydin Please let me double-check that with you - at the moment we are ok that the configuration supplied to MC is different from the one actually provided by a user for those instance-based properties, is that correct?
This is not the problem of this specific PR, because we used that approach before. But if the PR is merged, we will have more places where instance class is translated into a plain class name. I would like to ensure that it will not lead to any additional problems in MC.